src/main/java/com/google/sps/servlets/GetBlurAreasServlet.java

```diff
@@ -96,56 +128,93 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /**
-   * Uses the Google Cloud Vision API to find faces in the image represented by the binary data
-   * stored in @param imageBytes
+   * Uses the Google Cloud Vision API to find parts to blur in the image represented by the binary
+   * data stored in @param imageBytes
    *
-   * @return an ArrayList of bounding rectangles representing faces. Rectangles are represented by a
-   *     list of points. There is no guaranteed order of the points.
+   * @return an ArrayList of bounding rectangles representing parts to blur. Rectangles are
+   *     represented by a list of points. There is no guaranteed order of the points.
    */
-  private ArrayList<List<Point>> getBlurAreas(byte[] imageBytes) throws IOException {
+  private ArrayList<List<Point>> getBlurAreas(byte[] imageBytes, Integer partsToBlurMask)
+      throws IOException {
     // This is the array that we will return.
-    ArrayList<List<Point>> faceBorders = new ArrayList<>();
+    ArrayList<List<Point>> rectanglesToBlur = new ArrayList<>();
 
     // Convert bytes to an Image object.
     ByteString byteString = ByteString.copyFrom(imageBytes);
     Image image = Image.newBuilder().setContent(byteString).build();
 
-    // Create a face detection request to our image.
-    Feature feature = Feature.newBuilder().setType(Feature.Type.FACE_DETECTION).build();
-    AnnotateImageRequest request =
-        AnnotateImageRequest.newBuilder().addFeatures(feature).setImage(image).build();
+    // This will hold the image and the features we want to detect.
+    Builder requestBuilder = AnnotateImageRequest.newBuilder();
+
+    // Check which parts our bitmask contains and add the corresponding features to requestBuilder.
+    if ((partsToBlurMask & FACE_BLUR_MASK) != 0) {
+      Feature feature =
+          Feature.newBuilder().setType(Feature.Type.FACE_DETECTION).setMaxResults(0).build();
```

Comment:
The part 'setMaxResults(0)' is not clear. What does '0' mean in this case? Please add a comment, or put 0 into a variable that explains it.

Reply:
Right! I added a comment, thanks for pointing out!